presto-main/src/main/java/com/facebook/presto/cost/CoefficientBasedStatsCalculator.java

```diff
@@ -117,17 +116,8 @@ public PlanNodeStatsEstimate visitOutput(OutputNode node, Void context)
         @Override
         public PlanNodeStatsEstimate visitFilter(FilterNode node, Void context)
         {
-            PlanNodeStatsEstimate sourceStats;
-            if (node.getSource() instanceof TableScanNode) {
-                sourceStats = visitTableScanWithPredicate((TableScanNode) node.getSource(), node.getPredicate());
-            }
-            else {
-                sourceStats = visitSource(node);
-            }
-
-            final double filterCoefficient = FILTER_COEFFICIENT;
-            PlanNodeStatsEstimate filterStats = sourceStats
-                    .mapOutputRowCount(value -> value * filterCoefficient);
+            PlanNodeStatsEstimate sourceStats = visitSource(node);
```

Comment:
To make sure I understand.  We added complicated logic here because we didn't used to have the table layouts yet when the stats calculator was running.  But now that we run PickTableLayout earlier, we don't need this.

Reply:
This is correct. `TestCoefficientBasedStatsCalculator.testStatsCalculatorUsesLayout` still passes, showing the logic is no longer necessary.